### PR TITLE
feat(notification): Enable/Disable Endpoint

### PR DESCRIPTION
**Pull request by dmyios**
## Description

`PATCH /configuration/notifications/{notificationId}`

**Fixes** # MON-20076

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by jeremyjaouen — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

File: centreon/src/Core/Notification/Infrastructure/API/PartialUpdateNotification/PartialUpdateNotificationController.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        PartialUpdateNotification $useCase,
+        PartialUpdateNotificationPresenter $presenter,
+        int $notificationId
+    ): Response {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        try {
+            $data = $this->validateAndRetrieveDataSent($request, __DIR__ . '/PartialUpdateNotificationSchema.json');
+
+            $requestDto = new PartialUpdateNotificationRequest();
```
Comment (a-launois) [inline centreon/src/Core/Notification/Infrastructure/API/PartialUpdateNotification/PartialUpdateNotificationController.php:59]:
```suggestion
            /** @var array{is_activated?: bool} $requestDto */
            $requestDto = new PartialUpdateNotificationRequest();
```

File: centreon/doc/API/centreon-api-v23.10.yaml [hunk 2]
```diff
@@ -249,6 +249,8 @@ paths:
           $ref: '#/components/responses/InternalServerError'
   /configuration/notifications:
     $ref: "./v23.10/Configuration/Notification/Notifications.yaml"
```
Comment (a-launois) [inline centreon/doc/API/centreon-api-v23.10.yaml:251]: Forgot to update the "changelog" part

File: centreon/doc/API/v23.10/Configuration/Notification/PartialUpdateNotification.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+patch:
+  tags:
+    - Notification
+  summary: "Enable or disable a notification definition"
+  description: |
+    Enable or disable a notification definition
```
Comment (a-launois) [inline centreon/doc/API/v23.10/Configuration/Notification/PartialUpdateNotification.yaml:6]:
```suggestion
  summary: "Partially update a notification definition"
  description: |
    Partially update a notification definition
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/PartialUpdateNotificationRequest.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+type: object
+required: ["is_activated"]
```
Comment (a-launois) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/PartialUpdateNotificationRequest.yaml:2]: No properties are mandatory in a PATCH action

File: centreon/src/Core/Notification/Application/UseCase/PartialUpdateNotification/PartialUpdateNotification.php [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        }
+    }
+
+    /**
+     * @param PartialUpdateNotificationRequest $request
+     * @param Notification $notification
+     *
+     * @throws \Throwable
+     */
+    private function updateIsActivated(PartialUpdateNotificationRequest $request, Notification $notification): void
```
Comment (a-launois) [inline centreon/src/Core/Notification/Application/UseCase/PartialUpdateNotification/PartialUpdateNotification.php:149]: method's name is misleading as you technicaly update the entire notification

File: centreon/tests/api/features/Notification.feature [hunk 6]
```diff
@@ -486,3 +486,226 @@ Feature:
...
+
+    When I send a PATCH request to '/api/latest/configuration/notifications/1' with body:
+    """
+    {
+      "is_activated": false
+    }
+    """
+    Then the response code should be "204"
+
+  Scenario: Enable a notification as a non-admin user with insufficient rights
```
Comment (a-launois) [inline centreon/tests/api/features/Notification.feature:653]: Just a side note for another time: try to reduce the number of scenarios to improve CI performances ( I try to keep it to one scenario per endpoint when possible)

**Review by a-launois — CHANGES_REQUESTED** [review]

File: centreon/doc/API/centreon-api-v23.10.yaml (hunk 2 above)
Comment (dmyios) [reply centreon/doc/API/centreon-api-v23.10.yaml:252]: done

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/PartialUpdateNotificationRequest.yaml (hunk 4 above)
Comment (dmyios) [reply centreon/doc/API/v23.10/Configuration/Notification/Schema/PartialUpdateNotificationRequest.yaml]: done

File: centreon/src/Core/Notification/Application/UseCase/PartialUpdateNotification/PartialUpdateNotification.php (hunk 5 above)
Comment (dmyios) [reply centreon/src/Core/Notification/Application/UseCase/PartialUpdateNotification/PartialUpdateNotification.php]: done

**Review by a-launois — APPROVED** [review]

**Pull request merged**